geth/node/manager.go

```diff
@@ -31,86 +34,90 @@ var (
 	ErrRPCClient                   = errors.New("failed to init RPC client")
 )
 
+const (
+	started = 0
+	stopped = 1
+	pending = 2
+)
+
 // NodeManager manages Status node (which abstracts contained geth node)
+// uses interfaces to prevent races on exported objects.
 type NodeManager struct {
-	sync.RWMutex
-	config         *params.NodeConfig // Status node configuration
-	node           *node.Node         // reference to Geth P2P stack/node
-	nodeStarted    chan struct{}      // channel to wait for start up notifications
-	nodeStopped    chan struct{}      // channel to wait for termination notifications
-	whisperService *whisper.Whisper   // reference to Whisper service
-	lesService     *les.LightEthereum // reference to LES service
-	rpcClient      *rpc.Client        // reference to RPC client
+	state *int32
```

Comment:
Why is it a pointer? Also, it should be a separate type like `type state int`.

Reply:
We could use pointer or in all atomic operations we should do something like `atomic.StoreInt32(&...)`.

Reply:
And wouldn't it be more explicit to have it here as `state int32` and use `atomic.StoreInt32(&...)`?